### PR TITLE
Added functionalities for ProjCurveDivisor.

### PR DESCRIPTION
**Pull request by delphinepol**
Moved the files from examples to experimental
Modified the code from a previous pull request to make it more intrinsic,
added functions related to linear equivalence and principal divisor.

### CODE REVIEW

File: experimental/AffinePlaneCurve.jl [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+  L = Array{S, 1}[]
+  # For the linear factors of g, we add the constant coefficient to the list Y.
+  # The constant coefficient is stored as a mpoly to be used later in evaluate.
+  for g in keys(Z.fac)
+     if total_degree(g) == 1
+        f = g//lc(g)
+        push!(Y, -f + gen(R, 2))
+     end
+  end
+  if !isempty(Y)
```
Comment (fingolfin) [inline experimental/AffinePlaneCurve.jl]: This `if` is redundant: if `Y` is empty, the for loop `for y in Y` is not doing anything anyway.

File: experimental/AffinePlaneCurve.jl [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+     H = D.eq
+  end
+  I = ideal(R, [F, H])
+  # We eliminate the first variable in the ideal.
+  B = eliminate(I, [gen(R, 1)])
+  # We factorize the polynomial we obtain (which is a polynomial where only
+  # the second variable appears).
+  Z = factor(gen(B, 1))
+  Y = []
+  L = Array{S, 1}[]
```
Comment (fingolfin) [inline experimental/AffinePlaneCurve.jl]: Perhaps like this, for consistency?
```suggestion
  L = Vector{S}[]
```

File: experimental/AffinePlaneCurve.jl [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+     F = C.eq
+     H = D.eq
+  end
+  I = ideal(R, [F, H])
+  # We eliminate the first variable in the ideal.
+  B = eliminate(I, [gen(R, 1)])
+  # We factorize the polynomial we obtain (which is a polynomial where only
+  # the second variable appears).
+  Z = factor(gen(B, 1))
+  Y = []
```
Comment (fingolfin) [inline experimental/AffinePlaneCurve.jl]: This way, `Y` is a `Vector{Any}`. Perhaps we can say better and restrict the type?

File: experimental/AffinePlaneCurve.jl [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+     end
+  end
+  # L contains the intersection points as arrays (or can be empty).
+  Pts = Array{Point, 1}()
+  CC = Array{AffinePlaneCurve, 1}()
+  if !isempty(L)
+     for p in L
+        push!(Pts, Point(p))
+     end
+  end
```
Comment (fingolfin) [inline experimental/AffinePlaneCurve.jl]: I *think* the following would work
```suggestion
  Pts = Point[Point(p) for p in L]
  CC = Vector{AffinePlaneCurve}()
```

File: experimental/AffinePlaneCurve.jl [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+   D = reduction(C)
+   Pts = Array{Point, 1}()
+   CC = Array{AffinePlaneCurve, 1}()
+   # The components with multiplicity > 1 are singular
+   f = []
+   for (h, c) in comp
+      if c != 1
+         push!(f, h.eq)
+      end
+   end
```
Comment (fingolfin) [inline experimental/AffinePlaneCurve.jl]: This might also work (untested)
```suggestion
   f = [h.eq for (h,c) in comp if c != 1]
```

**Comment by fingolfin** [comment]
I was confused why this PR only seems to add stuff, when it ought to be renaming a bunch of things; I believe what happened is that you forgot to commit the removal of the old files. I've now added a commit to this PR which does that. A bunch of my comments now are outdated because they reference code that was already there, I just didn't notice because of the above

**Comment by fingolfin** [comment]
I've moved the files once more, to `experimental/PlaneCurve/`, which is more in line with the other content of the `experimental` dir (it makes is also obvious that those files belong together)

**Comment by delphinepol** [comment]
Thank you for moving the files, I didn't know that I had to remove the other ones... I moved the files to experimental as Tommy suggested me, and I edited also the file DivisorCurve by improving the code which was already there and adding some functions. I hope it was not too much in one pull request, sorry for the confusion...

**Pull request merged**